tree.go

```diff
@@ -1244,13 +1309,14 @@ func (n *LeafNode) updateLeaf(index byte, value []byte) error {
 
 	// If index is in the first NodeWidth/2 elements, we need to update C1. Otherwise, C2.
 	cxIndex := 2 + int(index)/(NodeWidth/2) // [1, stem, -> C1, C2 <-]
-	n.updateC(cxIndex, frs[0], frs[1])
+	n.updateC(cxIndex, frs[0], frs[1], curTs)
+	n.updateLastTs(curTs)
```

Comment:
`updateLastTs` doesn't update the commitment, so it should be called _before_ `updateC`